Lib/unittest/test/testmock/testthreadingmock.py

```diff
@@ -0,0 +1,173 @@
+import threading
+import multiprocessing
+import time
+import unittest
+import concurrent.futures
+
+from test.support import start_threads
+from unittest.mock import patch, ThreadingMock, call
+
+
+class Something:
+
+    def method_1(self):
+        pass
+
+    def method_2(self):
+        pass
+
+
+class TestThreadingMock(unittest.TestCase):
+
+    def _call_after_delay(self, func, /, *args, **kwargs):
+        time.sleep(kwargs.pop('delay'))
+        func(*args, **kwargs)
+
+
+    def setUp(self):
+        self._executor = concurrent.futures.ThreadPoolExecutor(max_workers=5)
+
+    def tearDown(self):
+        self._executor.shutdown()
+
+    def run_async(self, func, /, *args, delay=0, **kwargs):
+        self._executor.submit(self._call_after_delay, func, *args, **kwargs, delay=delay)
+
+    def _make_mock(self, *args, **kwargs):
+        return ThreadingMock(*args, **kwargs)
+
+    def test_instance_check(self):
+        waitable_mock = self._make_mock()
+
+        with patch(f'{__name__}.Something', waitable_mock):
+            something = Something()
+
+            self.assertIsInstance(something.method_1, ThreadingMock)
+            self.assertIsInstance(
+                something.method_1().method_2(), ThreadingMock)
+
+
+    def test_side_effect(self):
+        waitable_mock = self._make_mock()
+
+        with patch(f'{__name__}.Something', waitable_mock):
```

Comment:
For the reviewer: I left these tests with the patch function. All tests are using `patch` but I can change this to have a single test verifying patch and use the mock directly on the rest to simplify the test scenarios.